Automattic/jetpack
# Adds new endpoint for testing the connection to wpcom and UI for testing it while API is sandboxed

#### Changes proposed in this Pull Request:


This adds a new REST endpoint `jetpack/v4/connection/test` that is basically a wrapper for hitting wpcom's `/jetpack-blogs/%d/test-connection` endpoint. 

Also new is the ability to use this endpoint to check your connection to a sanboxed API, if applicable. 

It works like this: 
![test-connection-public](https://user-images.githubusercontent.com/7129409/47679059-aa649d80-db99-11e8-8b90-888bd2addba3.gif)


#### Testing instructions:





- Set the `JETPACK__SANDBOX_DOMAIN` to point to your wpcom sandbox
- You should see the badge in the masthead 
- Clicking the badge should fire a request to the new endpoint 
- Notices should be shown that reflect status
- Try clicking it when you are logged into your sandbox. It should succeed (if actually connected)
- Try clicking it when you have logged out of your sandbox. It should report as failed with the correct message. 

#### Proposed changelog entry for your changes:

- New REST API endpoint for testing the Jetpack connection.
